### PR TITLE
Raise exceptions in scanner tasks

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14942

---

With this patch, we get a generic error:

Screen Shot 2020-07-16 at 12 34 51

I added a kill switch too, in case something goes wrong...

Also, we have a retry strategy for the MAD call, so I didn't add anything. The MAD call could also be done later so it's less of a problem I think (for now).

### CODE REVIEW

File: src/olympia/scanners/tasks.py [hunk 1]
```diff
@@ -182,6 +182,10 @@ def run_yara(results, upload_pk):
        results as first argument.
     - `upload_pk` is the FileUpload ID.
     """
+    return _run_yara(results, upload_pk)
+
+
+def _run_yara(results, upload_pk):
```
Comment (bobsilverberg) [inline src/olympia/scanners/tasks.py:188]: What does this change (moving all the code into a separate function) do? Does the function where the exception is raised need to be *not* a `@validation_task`?

Comment (bobsilverberg) [reply src/olympia/scanners/tasks.py:188]: Oh, after looking at the tests I think this is just to support the tests, or maybe not?

Comment (willdurand) [reply src/olympia/scanners/tasks.py:188]: yes, exactly. `@validation_task` catches all exceptions so the new test cases wouldn't show that we actually raise exceptions there.

**Comment by willdurand** [comment]
> Is that why you requested review from both @diox and me?

Yes :)

**Pull request merged**